build/templates/_converters.py.mako

```diff
@@ -274,6 +274,73 @@ def convert_chained_repeated_capability_to_parts(chained_repeated_capability):
     return [','.join(collections.OrderedDict.fromkeys(x)) for x in repeated_capability_lists]
 
 
+def convert_resource_name(resource_name):
+    '''Convert a resource_name into a comma-separated list of device/channel resources
+
+    Args:
+        resource_name - Supported types:
+            - str - list (comma-delimited)
+            - str - single item
+            - list of str
+            - tuple of str
+            - None
+
+    Returns:
+        str - comma delimited string of each device/channel
+    '''
+    if resource_name is None:
+        return ''
+
+    return ','.join(_convert_resource_name(resource_name))
+
+
+@singledispatch
+def _convert_resource_name(arg):
+    raise errors.InvalidResourceNameError(str(arg), "Invalid Type")
+
+
+# This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
+@_convert_resource_name.register(str)  # noqa: F811
+def _(resource_name):
+    '''String version
+
+    We need to deal with a range ('Dev/0-3' or 'Dev/0:3'), a list ('Dev/0,Dev/1,Dev/2,Dev/3') and a single item
+    '''
+    # First we deal with a list
+    resource_name_list = resource_name.split(',')
+    if len(resource_name_list) > 1:
+        # We have a list so call ourselves again to let the iterable instance handle it
+        return _convert_resource_name(resource_name_list)
+
+    # Must be of form "Dev/0", "Dev/0:1", "Dev/0-1" or "Dev"
+    fields = [item.strip() for item in resource_name.split('/')]
+    device_name = fields[0]
+    if len(device_name) == 0:
```

Comment:
I was unclear whether this code should check for empty device name or if it should pass the empty device name to the driver to report the error.